Tanyel-Bariser/Vertex-Trigger
# Implement Rectanglular Path, improve MovingPlatform API and clean up LevelBuilders

- Implement RectanglePath using SimplePaths
- New factory and level builder methods
- CircularArray class
- Tests for above
- Create MovingPlatform class which relies on Paths for movement
- Create Path abstract class with one simple implementing class
- SimpleMovingPlatform supports horizontal and vertical Paths only
- Leave room to grow with more complex platform and path classes
- Move lots of utility code from level builder impls into the abstract
- class
- Clean up AbstractLevelBuilder, remove comments, optimize imports and
- rearrange methods
- Write unit test to verify SimplePath works as expected
- Add Rectangle Path to TanyelLevelBuilder
- Upgrade gradle and build tools version
- Add general android/libs exclusion to .gitignore
